src/fsharp/Optimizer.fs

```diff
@@ -2912,7 +2912,6 @@ and OptimizeLambdas (vspec: Val option) cenv env topValInfo e ety =
         let env = Option.foldBack (BindInternalValToUnknown cenv) baseValOpt env
         let env = BindTypeVarsToUnknown tps env
         let env = List.foldBack (BindInternalValsToUnknown cenv) vsl env
-        let env = BindInternalValsToUnknown cenv (Option.toList baseValOpt) env
```

Comment:
Is this line genuinely a no op?  Because for sure, I have no clue.

Reply:
It looks to be. They are both doing the name thing. It just adds it to a dictionary, but it's already there.